src/renderer/src/routes/pages/Send/index.tsx

```diff
@@ -1,114 +1,237 @@
+import { DoraHelper } from '@renderer/helpers/DoraHelper'
+import { useBsAggregatorSelector } from '@renderer/hooks/useBlockchainSelector'
+import { useModalNavigate } from '@renderer/hooks/useModalRouter'
+import { useEncryptedPasswordSelector } from '@renderer/hooks/useSettingsSelector'
 import { ContentLayout } from '@renderer/layouts/ContentLayout'
 
 import { Recipient } from './Recipient'
 import { SelectAccount } from './SelectAccount'
 import { SelectToken } from './SelectToken'
 import { SendAmount } from './SendAmount'
-import { TokenBalanceList } from './TokenBalanceList'
 import { TotalFee } from './TotalFee'
 
 enum SendPageStep {
   SelectAccount = 1,
   SelectToken = 2,
   SelectAmount = 3,
   SelectContact = 4,
+  Send = 5,
+}
+
+type TSendServiceResult = {
+  serviceAccount: Account
+  service: BlockchainService
+  token: TokenBalance
 }
 
 export const SendPage = () => {
   const { t } = useTranslation('pages', { keyPrefix: 'send' })
-  const { accounts } = useAccountsSelector()
-  const balanceExchange = useBalancesAndExchange(accounts[0])
   const [selectedAccount, setSelectedAccount] = useState<IAccountState>()
   const [selectedToken, setSelectedToken] = useState<TokenBalance | null>()
   const [selectedAmount, setSelectedAmount] = useState<number>(0)
-  const [currentStep, setCurrentStep] = useState<SendPageStep>(SendPageStep.SelectAccount)
+  const [selectedRecipient, setSelectedRecipient] = useState<string>('')
+  const [currentStep, setCurrentStep] = useState<number>(SendPageStep.SelectAccount)
+  const { modalNavigate } = useModalNavigate()
+  const [totalFee, setTotalFee] = useState<string>('0.00')
+  const { bsAggregatorRef } = useBsAggregatorSelector()
+  const { encryptedPasswordRef } = useEncryptedPasswordSelector()
 
-  const handleSelectAccount = account => {
+  const handleSelectAccount = (account: IAccountState) => {
     setSelectedAccount(account)
     setSelectedToken(null)
+    setSelectedAmount(0)
+    setSelectedRecipient('')
     setCurrentStep(SendPageStep.SelectToken)
   }
 
-  const handleSelectToken = token => {
+  const handleSelectToken = (token: TokenBalance) => {
     setSelectedToken(token)
+    setSelectedAmount(0)
+    setSelectedRecipient('')
     setCurrentStep(SendPageStep.SelectAmount)
   }
 
-  const handleSelectAmount = amount => {
+  const handleSelectAmount = (amount: number) => {
     setSelectedAmount(amount)
+    setSelectedRecipient('')
     setCurrentStep(SendPageStep.SelectContact)
   }
 
-  const defineStepColor = (step: SendPageStep) => {
-    if (currentStep < step) {
-      return 'text-gray-100'
+  const handleSelectRecipientAddress = (recipientAddress: string) => {
+    setSelectedRecipient(recipientAddress)
+    if (recipientAddress === '') {
+      setCurrentStep(SendPageStep.SelectContact)
+    } else {
+      setCurrentStep(SendPageStep.Send)
+    }
+  }
+
+  const populateSendService = useCallback(async (): Promise<TSendServiceResult | null> => {
```

Comment:
it think getSendService is a better name since this method is not populating anything